CDH2.py

```diff
@@ -0,0 +1,348 @@
+import os
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+from Bio.SeqUtils import gc_fraction
+from typing import Union, Tuple, Dict
+from abc import ABC, abstractmethod
+
+
+
+def analyse_gc(records: str, min_gc: Union[int, float], max_gc: Union[int, float]) -> str:
+    """
+    Return filtered FASTQ-sequences by GC-content
+    
+    :param records: FASTQ-file
+    :type records: str
+    :param min_gc: Left boundary for GC-content filtration
+    :type min_gc: Union[int, float]
+    :tupe max_gc: Right boundary for GC-content filtration
+    :type max_gc: Union[int, float]
+    :rtype: str
+    :return: filtered FASTQ-sequences
+    """
+    return {record.id: record for record in records if (min_gc/100) <= gc_fraction(record.seq) <= (max_gc/100)}
+
+
+def filter_by_length(records: str, min_length: Union[int, float], max_length: Union[int, float]) -> str:
+    """
+    Return filtered FASTQ-sequences by GC-content
+    
+    :param records: FASTQ-file
+    :type records: str
+    :param min_gc: Left boundary for GC-content filtration
+    :type min_gc: Union[int, float]
+    :tupe max_gc: Right boundary for GC-content filtration
+    :type max_gc: Union[int, float]
+    :rtype: str
+    :return: filtered FASTQ-sequences
+    """
+
+    return {record.id: record for record in records if min_length <= len(record.seq) <= max_length}
+
+
+def filter_by_quality(records: str, quality_threshold: Union[int, float]) -> str:
```

Comment:
Те же комментарии к аннотации типов к функциям ниже
```suggestion
def filter_by_quality(records: List[SeqRecord] quality_threshold: Union[int, float]) -> Dict[SeqRecord]:
```